src/Raven.Server/ServerWide/Maintenance/ClusterMaintenanceSupervisor.cs

```diff
@@ -375,10 +375,19 @@ private void UpdateNodeReportIfNeeded(ClusterNodeStatusReport nodeReport, List<D
                 // we take the last received and not the last successful.
                 // we don't want to reuse by mistake a successful report when we receive an 'unchanged' error.
                 var lastReport = ReceivedReport;
-                if (lastReport.Status != ClusterNodeStatusReport.ReportStatus.Ok)
+                switch (lastReport!.Status)
                 {
-                    throw new InvalidOperationException(
-                        $"We have databases with '{DatabaseStatus.NoChange}' status, but our last report from this node is '{lastReport.Status}'");
+                    case ClusterNodeStatusReport.ReportStatus.WaitingForResponse:
+                    case ClusterNodeStatusReport.ReportStatus.Timeout:
+                    case ClusterNodeStatusReport.ReportStatus.Error:
+                        throw new InvalidOperationException($"We have databases with '{DatabaseStatus.NoChange}' status, but our last report from this node is '{lastReport.Status}'");
+                    case ClusterNodeStatusReport.ReportStatus.Ok:
+                    case ClusterNodeStatusReport.ReportStatus.OutOfCredits:
+                    case ClusterNodeStatusReport.ReportStatus.EarlyOutOfMemory:
+                    case ClusterNodeStatusReport.ReportStatus.HighDirtyMemory:
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException();
```

Comment:
Maybe it's worth adding a message with information about what kind of status did we get?